### PR TITLE
Use https url for socket when page is loaded over htpps

### PR DESCRIPTION
**Pull request by nightwing**
This is useful In case when browser-sync works on `http://localhost` and its output is proxied to `htpps:` see http://stackoverflow.com/questions/33447135/cant-get-grunt-browser-sync-in-cloud9-ide-to-work for an example

### CODE REVIEW

**Comment by shakyShane** [comment]
Hi @nightwing thanks very much for your contribution

Could you provide a more concrete example? Browsersync can run in `https` without problem, so I may be missing your goal?

**Comment by nightwing** [comment]
Hi, thanks for checking this!

I have a node server that runs on `http://localhost:8080` on a vm
and access it via proxy as `https://workspace-user.c9users.io:8080`

If i try to use browserSync proxy as
```
    browserSync.init({
        proxy: 'localhost:8080',
        port: 8081
    });
```
it works on `http://workspace-user.c9users.io:8081`, but on `https://workspace-user.c9users.io:8081` socket connection fails.

When i try to add `https: true,` to the config browsersync tells me `Invalid config. You set https: true, but your proxy target doesn't reflect this. http://localhost:8081`, but even if this worked having to add that just for running on cloud9 would be too  cumbersome.

So basically what i want is to have
```objc
[http server]
   ||
   \/
[browsersync proxy]
   ||
   \/
[another proxy that can be either http or https]
```
and not have to change config for accommodating to the last proxy.

**Comment by HugoHeneault** [comment]
Any possiblity to see this merged? I'm needing it for a same case... Thanks! :)

**Comment by samuel-gay** [comment]
Agreed, this resolves an issue when using the Cloud9 IDE.

I noticed that when using the https url, chrome would block the web socket comms, because it was trying to connect over http. It meant that changes to updated css files were loaded but not applied in the website. It was quite confusing...

**Comment by shakyShane** [comment]
Thanks @nightwing - can you please try the master branch to verify it's working as you expect

```js
npm i browsersync/browser-sync
```

:)
